newdle/api.py

```diff
@@ -77,6 +81,38 @@ def require_token():
     g.user = user
 
 
+@api.route('/avatar/<email>')
+@allow_anonymous
+def user_avatar(email):
+    size = request.args.get('size')
+    email_hex = hashlib.md5(email.encode()).hexdigest()
+    # make gravat return 404 HTTP code instead of a default image
+    query_args = {'d': '404'}
+    if size is not None:
+        query_args['s'] = size
+    gravatar_url = (
+        f'https://gravatar.com/avatar/{quote(email_hex)}?' f'{urlencode(query_args)}'
```

Comment:
an md5 hexdigest is guaranteed to be only `[a-f0-9]`; no need for `quote()` there